### PR TITLE
Fix intersection for symbolics

### LINKED ISSUE #25222: `intersection` isn't correct in the symbolic case
```python
>>> Plane((x,y,z),normal_vector=(1,1,1)).intersection(Line((x,y,z),(1,-1,0)))
[Point3D(x, y, z)]
>>> Plane((0,0,0),normal_vector=(1,1,1)).intersection(Line((0,0,0),(1,-1,0)))
[Line3D(Point3D(0, 0, 0), Point3D(1, -1, 0))]
```
So the first line should really return `Piecewise(([Line3D(Point3D(0, 0, 0), Point3D(1, -1, 0))],And(Eq(x,0),Eq(y,0),Eq(z,0))),([Point3D(x, y, z)],True))`.

Logic behind this is discussed [here](https://github.com/sympy/sympy/pull/25220#discussion_r1223525099).

### PR DESCRIPTION
**Pull request by madt2709**
#### References to other Issues or PRs

Fixes #25222

#### Brief description of what is fixed or changed
`Plane.intersection` now supports symbolic case by returning `Piecewise` object. It effectively replicates the old logic but turns the conditions into equations to satisfy. 

#### Other comments
Currently, the plane intersecting another plane case won't work. This is due to https://github.com/sympy/sympy/issues/25228. This function is called when assessing `if self.equals(o)`. 

I will make a quick fix for the above and then we should re run the tests on this PR.

#### Release Notes

* geometry
  * Added symbolic case to `Plane.intersection`. Previously, the results returned were wrong in the symbolic case.

### CODE REVIEW

File: sympy/geometry/plane.py [hunk 1]
```diff
@@ -384,11 +386,15 @@ def intersection(self, o):
...
         """
+        symbolic_self = [any(isinstance(i, Symbol) for i in self.p1), any(isinstance(i, Symbol) for i in self.normal_vector)]
         if not isinstance(o, GeometryEntity):
             o = Point(o, dim=3)
         if isinstance(o, Point):
             if o in self:
                 return [o]
+            # Check for symbolic case.
+            elif any(isinstance(i, Symbol) for i in o) or any(i for i in symbolic_self):
+                return [Piecewise((o,Eq((o-self.p1).dot(Point3D(self.normal_vector)),0)),(set(),True))]
```
Comment (1e9abhi1e10) [inline sympy/geometry/plane.py:397]: Should be a space after a comma

**Comment by 1e9abhi1e10** [comment]
Looks like it is your first PR, you need to add your name and email in the `.mailmap` file.
Check [instructions](https://docs.sympy.org/dev/contributing/new-contributors-guide/workflow-process.html#add-your-name-and-email-address-to-the-mailmap-file).

**Comment by madt2709** [comment]
@1e9abhi1e10 I would like for https://github.com/sympy/sympy/pull/25231 to be merged first. I added my author name on that PR instead. Once https://github.com/sympy/sympy/pull/25231 is merged than we can merge this PR.

File: sympy/geometry/plane.py (hunk 1 above)
Comment (madt2709) [reply sympy/geometry/plane.py:397]: Good spot, will update

**Comment by madt2709** [comment]
@1e9abhi1e10 we've opted only to raise an error message in case of a clash of variables. Therefore, I have updated the variable names in the tests and added my author name here. Please approve if you think this can go in 🙏

**Pull request closed without merging**